### PR TITLE
Dedup throwodspnetworkerror in odsp driver and odsp doclib utils.

### PR DESCRIPTION
**Pull request by jatgarg**

### CODE REVIEW

**Review by AndreiZe — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspUtils.ts
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspUtils.ts:133]: Actually typescript is complaining that we don't have return statement in the end. So need to put it here outside of catch and then add assert otherwise we will need to add undefined in return type.

Comment (markfields) [reply packages/drivers/odsp-driver/src/odspUtils.ts:133]: Ok obviously there's already a `createOdspNetworkError`. It's kind of like we have 2 overloads for that function, if we can merge them with a coherent signature, or name them well but have them both return the error instead of one also throwing I think that's better.  Your call.

File: packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts
Comment (markfields) [inline packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts:175]: I can't find any usages of this... remove it and on the other function?

**Review by markfields — APPROVED** [review]
🎂 I did leave some comments for little improvements here and there, but feel free to merge and follow up in another PR to unblock Andrei if you want.

File: packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts
Comment (jatgarg) [reply packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts:175]: Actually this is added just recently. So may not be yet in telemetry. So lets keep it.

File: packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts
Comment (jatgarg) [reply packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts:163]: Yeah, lets keep it. MIght be useful to know anything about error that spo throws.

File: packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts
Comment (jatgarg) [reply packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts:159]: done

File: packages/utils/odsp-doclib-utils/package.json
Comment (jatgarg) [reply packages/utils/odsp-doclib-utils/package.json:14]: Yes. I was seeing some warning about the commands being wrong. I don't know about the tool. It is maybe the curtis' scripts detecting this.

File: packages/drivers/odsp-driver/src/odspError.ts
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspError.ts:10]: maybe move errorObjectFromSocketError (that follows below) also with it?

File: packages/drivers/odsp-driver/src/odspUtils.ts
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspUtils.ts:131]: The throwOdspNetworkError  actually also extracts message from response object and then adds to this custom message. This is just to add any custom message so that it is easier to make messages more meaningful as sometimes the error messages from spo are not so meaningful. I think we can trust that the error thrown by spo apis do not contain any PII info.

File: packages/drivers/odsp-driver/src/odspUtils.ts
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspUtils.ts:133]: Instead of moving return statement around, I think it would be more readable (and less lines of code) to leave as is but add
> assert(false, "unreached");
statement at the end
Even better would be to mark throwOdspNetworkError() as returning never, and that should remove the need to change anything here.

**Review by vladsud — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspUtils.ts
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspUtils.ts:133]: done

File: packages/drivers/odsp-driver/src/odspError.ts
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspError.ts:10]: Lets keep it here until some need arises. Actually throwodspNetworkError was duplicated, so we thought of moving it. IOdsoNetworkError which is used here is also defined in contracts. So for now lets keep it here.

**Pull request merged**
